### PR TITLE
Add step detection to sAHP

### PR DESCRIPTION
**Pull request by DrTaDa**

### CODE REVIEW

File: bluepyefe/ecode/sAHP.py [hunk 1]
```diff
@@ -109,10 +104,98 @@ def interpret(self, t, current, config_data, reader_data):
...
 
-        # Converting back to ms
-        for name_timing in ["ton", "tmid", "tmid2", "toff"]:
-            self.index_to_ms(name_timing, t)
-        self.tend = len(t) * self.dt
+    def step_detection(self, current, config_data, reader_data):
+
+        # Set the threshold to detect the step
+        noise_level = numpy.std(numpy.concatenate((self.current[:50], self.current[-50:])))
+        step_threshold = numpy.clip(4.5 * noise_level, 1e-5, numpy.max(self.current))
```
Comment (AurelienJaquier) [inline bluepyefe/ecode/sAHP.py:111]: Why 4.5 specifically?

File: bluepyefe/ecode/sAHP.py [hunk 2]
```diff
@@ -109,10 +104,98 @@ def interpret(self, t, current, config_data, reader_data):
...
+        smooth_current = scipy_signal2d(current, 85)
+
+        # Infer the beginning of the long step
+        self.hypamp = base_current(current)
+
+        if "ton" in config_data and config_data["ton"] is not None:
+            self.ton = int(round(config_data["ton"] / self.dt))
+        else:
+            _ = numpy.abs(smooth_current[idx_buffer:] - self.hypamp)
+            self.ton = idx_buffer + numpy.argmax(_ > step_threshold)
```
Comment (AurelienJaquier) [inline bluepyefe/ecode/sAHP.py:129]: Can't you find a better variable name instead of '_'?

File: bluepyefe/ecode/sAHP.py [hunk 3]
```diff
@@ -109,10 +104,98 @@ def interpret(self, t, current, config_data, reader_data):
...
+
+        # Infer the end of the long step
+        _ = numpy.flip(
```
Comment (AurelienJaquier) [inline bluepyefe/ecode/sAHP.py:132]: Same here, I feel like naming variables _ and using them afterwards is not optimal for code reading comprehension

File: bluepyefe/ecode/sAHP.py [hunk 4]
```diff
@@ -109,10 +104,98 @@ def interpret(self, t, current, config_data, reader_data):
...
+        )
+
+        # Get the amplitude of the step current (relative to hypamp)
+        self.amp = numpy.median(
+            numpy.concatenate((smooth_current[self.ton:self.ton + 50],
+                               smooth_current[self.toff - 50:self.toff]))
+        ) - self.hypamp
+
+        # Infer the beginning of the short step
+        _ = numpy.abs(
```
Comment (AurelienJaquier) [inline bluepyefe/ecode/sAHP.py:146]: no _ pls

File: bluepyefe/ecode/sAHP.py [hunk 5]
```diff
@@ -109,10 +104,98 @@ def interpret(self, t, current, config_data, reader_data):
...
+            smooth_current[self.ton + buffer_step:self.toff - buffer_step] -
+            self.amp - self.hypamp
+        )
+        self.tmid = self.ton + buffer_step + numpy.argmax(_ > step_threshold)
+
+        # Infer the end of the long step
+        _ = numpy.flip(
```
Comment (AurelienJaquier) [inline bluepyefe/ecode/sAHP.py:153]: _

File: bluepyefe/ecode/sAHP.py (hunk 1 above)
Comment (DrTaDa) [reply bluepyefe/ecode/sAHP.py:111]: It was the value that worked for all the data I had available. Above and below it misses some steps.

**Review by AurelienJaquier — APPROVED** [review]
Alright, looks good to me!

**Pull request merged**